tracer.go

```diff
@@ -182,7 +182,7 @@ func (t *tracer) startSpanWithOptions(
 
 	var samplerTags []tag
 	var ctx SpanContext
-	debugRequest := (hasParent && parent.isDebugIDContainerOnly())
+	debugRequest := (hasParent && parent.isContainerOnly())
```

Comment:
come to thing of it, I don't believe the trace must always be sampled/debugged just because it has baggage. I would separate these two. Baggage will propagate even if the trace is not sampled.